hs/src/Reach/Optimize.hs

```diff
@@ -336,8 +336,8 @@ instance Optimize DLExpr where
     DLE_TimeOrder at tos -> DLE_TimeOrder at <$> opt tos
     DLE_GetContract at -> return $ DLE_GetContract at
     DLE_GetAddress at -> return $ DLE_GetAddress at
-    DLE_EmitLog at m a -> DLE_EmitLog at m <$> opt a
-    DLE_setApiDetails at w t c -> return $ DLE_setApiDetails at w t c
+    DLE_EmitLog at m ma a -> DLE_EmitLog at m ma <$> opt a
+    DLE_setApiDetails {..} -> return $ DLE_setApiDetails {..}
```

Comment:
You should write it out so that if we ever add more we will be forced to think about if they need to be freshened